### ISSUE TITLE
[ Skimming ] Barbershop: GAN-based Image Compositing using Segmentation Masks

### ISSUE THREAD

**Issue by doublejy715** [doublejy715/Paper_review#35]

**Comment by doublejy715** [comment]
# 1. Introduction
본 논문은 여러 reference image를 이용하여 image editing을 가능하게 하는 새로운 도구를 소개한다. 특히 hair editing에 특화되어 있다.

최근 latent space manipulation 기반으로 face editing 하는 모델들은 pose, expression, gender, age같은 속성들을 변화시킬 수 있었음.

다른 방식으로는 source image의 얼굴 속성을 target image에 부여하는 방식으로 image editing을 시도하였음.

hair transfer and face-swapping은 최근 많이 시도되는 image editing 이지만 매우 어려운 문제임. 속성을 가져올 때 빛, 얼굴, 옷, 배경 등의 여러 요소가 속성에 영향을 미치기 때문에 다른 이미지에서 얼굴 속성을 가져오면 부자연스럽게 보일 수 있음.

**Comment by doublejy715** [comment]
# 3. Method
## 3.1. Overview
reference images의 semantic region을 선택해서 원하는 이미지를 합성가능함.
> 여기서는 자동으로 reference image에 segmentation을 하고 target semantic segmentation mask image M을 생성한다.

hairstyle transfer를 위해서 한 이미지에서는 hairstyle을, 다른 이미지에서는 semantic categories 정보를 가져온다.

K개의 reference image가 있다면(I_{k}) 으로 가정한다. 그리고 새로운 이미지를 생성하기 위해서 reference image에서 원하는 요소를 가져와 target mask에 align하고 blend한다.(생성된 이미지를 I^{blend}라 한다.)

blended image의 latent code를 찾기 위해서 소개하는 단락

**Comment by doublejy715** [comment]
## 3.2 Initial Segmentation
### target mask 생성
reference image들을 segment 한 다음에 원하는 얼굴 요소를 복사해서 target image에 붙여넣는다.
M_{k} = Segment(I_{k}) (BiSeNET을 이용하여 segmentation 진행하였음)
> mask를 만들때 자동으로 reference image에서 그대로 가져올 수 있지만 수동으로 직접 마스크를 수정해도 된다.

target segmentation mask M 영역과 동일한 위치에 얼굴 요소가 존재하는 새로운 이미지 I^{blend}를 만드는 것이 주요 목적이다.(단순히 I^{blend}와 target segmentation mask M은 pair라 보면 편할 듯)
 I^{blend}는 reference image I_{k}로 부터 transfer 하여 생성된다. 
> M(x,y) = k : mask의 x,y 픽셀 위치에는 reference image의 순번이 기록되어 있다고 가정한다.

> translate가 잘 되려면?  
> 얼굴이 이미지 중심에 있어야 한다. 정렬 관련해서는 생략함

**Comment by doublejy715** [comment]
## 3.3 Embedding
먼저 reference image들을 mask 머리 형태에 적합한 이목구비를 가진 새로운 reference image를 align해서 target mask M와 일치시킨다. 본 논문에서는 align 방식을 두가지로 분류했음
(1) Reconstruction : input image I_{k}를 recon하기 위해서 latent code C^{rec}_{k}를 찾아낸다.
(2) Alignment : target mask M과 생성된 이미지 간 cross-entropy 값을 최소화시켜 C^{align}_{k}를 찾는다.

### 3.3.1 Reconstruction
input image I_{k}에서 reconstruction code C^{rec}_{k}를 찾는 방법을 소개한다. 이후 G(C^{rec}_{k})를 이용해 I_{k}를 만들어 낸다.(G는 stylegan2의 synthesis network임)

II2S를 시작하기 위한 reconstuction code C^{rec}_{k} 찾아내고, II2S는 StyleGAN2의 W+ latent-space에 embedding된 latent code w^{rec}_{k} 를 찾아준다.
> C^{rec}_{k}  -> [ II2S ] -> w^{rec}_{k} -> [ G ] -> I_{k}

#### W / W+ & II2S
reconstruction algorithm은 reconstruction quality 와 suitability간 trade-off를 통해서 editing이나 image compositing이 이루어진다.
W latent space는 512개의 vector만 존재하므로 디테일한 부분까지 묘사하기 어렵다. W+ latent space와 II2S를 이용하면 latent space의 표현력이 증가하지만, 여전히 괜찮은 reconstruction을 만들어 내기엔 역부족 이었음.

#### FS space
noise embedding을 이용하면 거의 완벽한 reconstruction 이 가능하지만 dataset의 artifact에 overfitting 되어 recon, edit에는 부적합 할 수 있음.

noise embedding 없이 W+ space 보다 더 쉽게 control 가능함. 또한 새로운 이미지를 design 하는 것도 가능.

**Comment by doublejy715** [comment]
# 4. Results
TITAN Xp graphics card * 1사용
- II2S embedding : take 2 min per image
- find recon C latent code : take 1 min per image
- generate align image : take 2 min per image
- generate L_{mask} : take 2 min per image

**Comment by doublejy715** [comment]
# 4. Results
TITAN Xp graphics card * 1사용
- II2S embedding : take 2 min per image
- find recon C latent code : take 1 min per image
- generate align image : take 2 min per image
- generate L_{mask} : take 2 min per image

## 4.1 Dataset
- image size : 1024x1024
- 
- target segmentation mask : generated automatically

## 4.2 Competing methods
MichiGAN, LOHO

## 4.3 Comparsion
### 4.3.1 User Study
- Amazon Mechanical Turk로 user study 진행
- segmentation from CelebAMask-HQ (19 category)
- hairstyle : CelebAMask-HQ의 hair 부분 사용
- Identity image : CelebAMask-HQ의 hair 이외의 영역 사용
- 비교 대상 : MichiGAN, LOHO -> checked in Fig 8
- 합성 퀄리티를 질문하였을 때 : our approach good

### 4.3.2. Reconstruction Quality
reconstruction quality of an embedding를 측정하기 위해서 RMSE, PSNR, SSIM, VGG perceptual similarity, LPIPS perceptual similarity, FID score를 이용하였음. table 1을 참조하면 됨

## 4.4 Ablation Study
Fig 9 를 통해서 hair transfer 정로를 가능해 볼 수 있음

### first column
- w/o FS latent space & w/o image alignment -> W+ w/o Align
- reference image에서 머리카락이 있던 disoccluded 영역이 사라지지 않음.
### second column
- F^{rec}_{k}를 사용하였음.
- original image의 detail한 부분을 유지할 수 있었음.
- reference image에서 보이지 않는 부분은 hair가 생성되지 않음.
### third column
- F^{align}_{k} 도 포함하였음. w/o FS space
- original face의 특징을 잃어버리고 일반적인 얼굴을 생성함. 

## 4.5 Qualitative Results
본 방법을 사용하였을 때 다양하고 괜찮은 결과를 얻을 수 있는 방법을 설명한다.
- Fig 6을 보면 다양한 머리 속성을 가져올 수 있음
- 사용자가 직접 마스크를 수정하고, reference image에서 segmented region이나 structure information을 가져올 수 있음.
- Fig 7을 보면 얼굴만이 아니라 다른 부분도 좋은 퀄리티로 edit 가능함을 보임.
- Fig 8. 다른 sota와 비교하는 부분. 머리 가장자리와 배경이 잘 어울리도록 마무리한 모습, reference image간 misalignment로 인해서 발생하는 artifact를 해결하였음. lighting을 고려한 transfer도 고려함
- 반면 본 방법은 hair editing에 초점을 맞췄기 때문에 배경을 재생성 하는데 미숙함을 보임.

## 4.6 Limitations
1. FS latent space를 통해서 recon 능력을 올렸지만, 여전히 보석, 장식품 같은 물체를 생성하는데 한계점이 보임. Fig 10(2, 4)
2. occlusion이 발생하면 이상한 결과를 가져다 줌. Fig 10(3, 5) 부분적으로 얇은 머리카락은 잡아내기 힘들어 그냥 얼굴이 보임
3. reference image와 target segmentation mask가 완전하게 overlap되지 않으면 aligning embedding이 잘 안됨
4. structure tensor가 target mask에 일치하도록 해야 Fig10(6) 현상이 덜생긴다.
